main.tf

```diff
@@ -1 +1,381 @@
+    ]
+    effect    = "Allow"
+    resources = [aws_sns_topic.eks_sns_topic.arn]
+  }
+
+  statement {
+    sid = "FirehosePermissions"
+    actions = [
+      "firehose:DescribeDeliveryStream"
+    ]
+    effect    = "Allow"
+    resources = [aws_kinesis_firehose_delivery_stream.extended_s3_stream.arn]
+  }
+
+  statement {
+    sid       = "AccountAliasPermissions"
+    actions   = ["iam:ListAccountAliases"]
+    effect    = "Allow"
+    resources = ["*"]
+  }
+
+  statement {
+    sid       = "Debug"
+    resources = ["*"]
+    effect    = "Allow"
+    actions = [
+      "eks:ListClusters",
+      "logs:DescribeLogGroups",
+      "firehose:ListDeliveryStreams",
+      "sns:ListTopics",
+      "s3:ListAllMyBuckets",
+      "s3:GetBucketAcl",
+      "s3:GetBucketLocation",
+      "s3:GetBucketNotificationConfiguration",
+      "cloudwatch:GetMetricData"
+    ]
+  }
+}
+
+resource "aws_iam_role" "eks_cw_iam_role" {
+  name               = local.eks_cw_iam_role_name
+  assume_role_policy = data.aws_iam_policy_document.eks_cw_assume_role_policy.json
+  depends_on         = [module.lacework_eks_audit_iam_role, aws_iam_role_policy_attachment.eks_cross_account_role_policy]
+}
+
+data "aws_iam_policy_document" "eks_cw_assume_role_policy" {
+  version = "2012-10-17"
+
+  statement {
+    sid     = "LaceworkEKSCloudWatchIAMRole"
+    actions = ["sts:AssumeRole"]
+    effect  = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = [local.log_region]
+    }
+  }
+}
+
+resource "aws_iam_policy" "eks_cw_iam_policy" {
+  name        = local.cw_iam_policy_name
+  description = "EKS Cloudwatch IAM policy"
+  policy      = data.aws_iam_policy_document.eks_cw_iam_role_policy.json
+}
+
+data "aws_iam_policy_document" "eks_cw_iam_role_policy" {
+  version = "2012-10-17"
+
+  statement {
+    actions = [
+      "firehose:*",
+    ]
+    effect    = "Allow"
+    resources = ["arn:aws:firehose:*:${data.aws_caller_identity.current.account_id}:*"]
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "eks_cw_iam_role_policy" {
+  role       = local.eks_cw_iam_role_name
+  policy_arn = aws_iam_policy.eks_cw_iam_policy.arn
+  depends_on = [aws_iam_role.eks_cw_iam_role, aws_iam_policy.eks_cw_iam_policy]
+}
+
+// for_each cluster, create a subscription filter
+resource "aws_cloudwatch_log_subscription_filter" "lacework_eks_cw_subscription_filter" {
+  for_each        = toset(var.cluster_names)
+  name            = "${var.prefix}-${each.value}-eks-cw-${random_id.uniq.hex}"
+  role_arn        = aws_iam_role.eks_cw_iam_role.arn
+  log_group_name  = "/aws/eks/${each.value}/cluster"
+  filter_pattern  = "{ $.stage = \"ResponseComplete\" && $.requestURI != \"/version\" && $.requestURI != \"/version?*\" && $.requestURI != \"/metrics\" && $.requestURI != \"/metrics?*\" && $.requestURI != \"/logs\" && $.requestURI != \"/logs?*\" && $.requestURI != \"/swagger*\" && $.requestURI != \"/livez*\" && $.requestURI != \"/readyz*\" && $.requestURI != \"/healthz*\" }"
```

Comment:
where did we get this filter from 😮 - this is hard to read and understand. Either I trust this is what we want or, you show me where this is coming from so I can review 😅